### PR TITLE
Improve iptables tests

### PR DESCRIPTION
**Pull request by howardjohn**
This changes tests to follow a standard golden file test pattern. This makes seeing diffs, seeing expected output, making changes, etc all easier. Additionally, a ton of test code is deduplicated.

The biggest change here is we always call `run()` instead of some tests which called targeted functions. IMO this is a better testing methodology, and the increased test output sizes are worth it to keep things simpler, more realistic, and detect any unexpected changes.

For some of the tests, I modified the job to write the `expected []string` content to the golden file, to ensure the new test is identical. For a few that did non-standard things, I spot checked the result to ensure they were the same.

### CODE REVIEW

**Comment by howardjohn** [comment]
Motivation is to help test https://github.com/istio/istio/issues/33965

File: tools/istio-iptables/pkg/builder/iptables_builder_impl.go [hunk 1]
```diff
@@ -34,22 +34,22 @@ type Rules struct {
 	rulesv6 []*Rule
 }
 
-// IptablesBuilderImpl is an implementation for IptablesBuilder interface
-type IptablesBuilderImpl struct {
+// IptablesBuilder is an implementation for IptablesBuilder interface
```
Comment (hzxuzhonghu) [inline tools/istio-iptables/pkg/builder/iptables_builder_impl.go:37]: The comments should be changed to `adding iptables rules`

And may rename the file name to iptables_builder.go

**Review by hzxuzhonghu — COMMENTED** [review]
lgtm. i like this style

**Review by ramaraochavali — COMMENTED** [review]
like this approach. Gold files definitely give better idea.

**Review by bianpengyuan — APPROVED** [review]
Verified that there is no lost of test coverage. Thanks!

**Pull request merged**